test/tests/functional/pbs_basil_support.py

```diff
@@ -712,6 +720,7 @@ def test_job_run_on_knl_node(self):
             self.logger.info("exec_vnode %s is a KNL vnode." % (evnode))
             rv = True
         else:
+            rv = False
             self.logger.info("exec_vnode %s is not a KNL vnode." % (evnode))
```

Comment:
This works, and feel free to leave it, but you could do this another way.  Instead of the if evnode in klist, you can say self.assertTrue(evnode in klist, false_msg).  You follow this immediately by the true logger message.  It's 2 lines instead of 7 lines.

Reply:
For now I would like to keep it as it is .